modules/core/src/test/scala/fs2/kafka/KafkaConsumerSpec.scala

```diff
@@ -352,4 +353,39 @@ final class KafkaConsumerSpec extends BaseKafkaSpec {
       consumed should contain theSameElementsAs produced.drop(readOffset.toInt)
     }
   }
+
+  describe("KafkaConsumer#assignmentStream") {
+    it("should stream assignment updates to listeners") {
+      withKafka { (config, topic) =>
+        createCustomTopic(topic, partitions = 3)
+
+        val consumer = (queue: NoneTerminatedQueue[IO, SortedSet[TopicPartition]]) =>
+          consumerStream[IO]
+            .using(consumerSettings(config))
+            .evalTap(_.subscribeTo(topic))
+            .evalMap { consumer =>
+              consumer.assignmentStream
+                .evalMap(as => queue.enqueue1(Some(as)))
+                .compile
+                .drain
+                .start
+                .void
+            }
+
+        (for {
+          queue <- Stream.eval(Queue.noneTerminated[IO, SortedSet[TopicPartition]])
+          _ <- consumer(queue)
+          _ <- Stream.eval(IO.sleep(5.seconds))
+          _ <- consumer(queue)
+          _ <- Stream.eval(IO.sleep(5.seconds))
+          _ <- Stream.eval(queue.enqueue1(None))
+          allUpdates <- Stream.eval(queue.dequeue.compile.toList)
+          _ <- Stream.eval(IO(assert {
+            println(allUpdates)
+            allUpdates.length == 3
```

Comment:
I'm thinking 3 because I'd expect:
1. Initial assignment of first consumer
2. Initial assignment of second consumer
3. Rebalanced set from first consumer, after the second joins

What I'm seeing instead is a list of two empty sets. Something is _really_ off because the `println`s in the main code show that we're hitting the `Init` and `Update` branches for both consumers, so we should be enqueuing 4 things.